march_hardware_builder/src/hardware_builder.cpp

```diff
@@ -210,24 +217,36 @@ std::unique_ptr<march::TemperatureGES> HardwareBuilder::createTemperatureGES(con
 
   const auto slave_index = temperature_ges_config["slaveIndex"].as<int>();
   const auto byte_offset = temperature_ges_config["byteOffset"].as<int>();
-  return std::make_unique<march::TemperatureGES>(march::Slave(slave_index, pdo_interface, sdo_interface), byte_offset);
+  for (auto ges : this->ges_list_)
+  {
+    if (ges->getSlaveIndex() == slave_index)
+    {
+      return ges;
+    }
+  }
+
+  std::shared_ptr<march::TemperatureGES> ges =
+      std::make_unique<march::TemperatureGES>(march::Slave(slave_index, pdo_interface, sdo_interface), byte_offset);
+  this->slave_list_.push_back(ges);
+  this->ges_list_.push_back(ges);
+  return ges;
 }
 
-std::unique_ptr<march::PowerDistributionBoard> HardwareBuilder::createPowerDistributionBoard(
-    const YAML::Node& pdb, march::PdoInterfacePtr pdo_interface, march::SdoInterfacePtr sdo_interface)
+std::shared_ptr<march::PowerDistributionBoard> HardwareBuilder::createPowerDistributionBoard(
```

Comment:
Why has this been changed to a `shared_ptr`?

Reply:
Same